docs/API.md

```diff
@@ -0,0 +1,275 @@
+
+If `databaseName` is provided, it should be the name of an attached database (or `"main"`). This causes only that database to be checkpointed.
+
+If the checkpoint fails, an `Error` is thrown.
+
+### .register([*options*], *function*) -> *this*
+
+Registers the given `function` so that it can be used by SQL statements.
+
+```js
+db.register(function add2(a, b) {return a + b;});
+db.prepare('SELECT add2(?, ?)').get(12, 4); // => 16
+db.prepare('SELECT add2(?, ?)').get('foo', 'bar'); // => 'foobar'
+db.prepare('SELECT add2(?, ?, ?)').get(12, 4, 18); // => Error: wrong number of arguments
+```
+
+By default, registered functions have a strict number of arguments (determined by `function.length`). You can register multiple functions of the same name, each with a different number of arguments, causing SQLite3 to execute a different function depending on how many arguments were passed to it. If you register two functions with same name and the same number of arguments, the second registration will erase the first one.
+
+If `options.name` is given, the function will be registered under that name (instead of defaulting to `function.name`).
+
+If `options.varargs` is `true`, the registered function can accept any number of arguments.
+
+If your function is [deterministic](https://en.wikipedia.org/wiki/Deterministic_algorithm), you can set `options.deterministic` to `true`, which may improve performance under some circumstances.
+
+```js
+db.register({name: "void", deterministic: true, varargs: true}, function () {});
+db.prepare("SELECT void()").get(); // => null
+db.prepare("SELECT void(?, ?)").get(55, 19); // => null
+```
+
+You can create custom [aggregates](https://sqlite.org/lang_aggfunc.html) by using generator functions. Your generator function must `yield` a regular function that will be invoked for each row passed to the aggregate.
+
+```js
+db.register(function* addAll() {
+  var total = 0;
+  yield function (rowValue) {total += rowValue;};
+  return total;
+});
+var totalTreasure = db.prepare('SELECT addAll(treasure) FROM dragons').pluck().get();
+```
+
+### .loadExtension(*path*) -> *this*
+
+Loads a compiled [SQLite3 extension](https://sqlite.org/loadext.html) and applies it to the current database connection.
+
+It's your responsibility to make sure the extensions you load are compiled/linked against a version of [SQLite3](https://www.sqlite.org/) that is compatible `better-sqlite3`. Keep in mind that new versions of `better-sqlite3` will periodically use newer versions of [SQLite3](https://www.sqlite.org/). You can see which version is being used [here](https://github.com/JoshuaWise/better-sqlite3/tree/master/deps).
```

Comment:
The link here could be changed to a relative link too (https://github.com/JoshuaWise/better-sqlite3/tree/master/deps)